DatadogCore/Sources/Utils/CoreMetrics.swift

```diff
@@ -92,3 +93,23 @@ internal enum BatchDeletedMetric {
         }
     }
 }
+
+/// Definition of "Batch Closed" telemetry.
+internal enum BatchClosedMetric {
```

Comment:
non blocking, but would have preferred this as struct with method `toAttributes() -> expected type` which makes it easier handle any conversion.

Reply:
I have been thinking of it, but I decided to keep it bare simple, as we introduce a new concept here and not yet sure which direction it will go. Moving from "static constants" model to a "type" won't cost much in the future.

For instance, one idea I had was to define `Metric` protocol that enforces common fields: `name`, `metric_type` and `attributes`. That could be passed directly to `telemetry.metric(_:)` in a type-safer way, instead of broad `telemetry.metric(name:attributes:)`.

With few options in hand, I bet on simplicity for now, so will keep it as it is.